### PR TITLE
[subset] Add gsub closure.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: src/hb-subset-input.cc [hunk 1]
```diff
@@ -117,3 +118,19 @@ hb_subset_input_drop_hints (hb_subset_input_t *subset_input)
...
+
+/**
+ * hb_subset_input_drop_ot_layout:
+ * @subset_input: a subset_input.
+ *
+ * If enabled ot layout tables will be dropped as part of
+ * the subsetting operation. Currently this defaults to
+ * true.
+ *
+ * Since: 1.8.0
```
Comment (ebraminio) [inline src/hb-subset-input.cc]: You can use REPLACEME here instead

Comment (garretrieger) [reply src/hb-subset-input.cc]: Thanks, done.

File: src/hb-subset-input.cc [hunk 2]
```diff
@@ -117,3 +118,19 @@ hb_subset_input_drop_hints (hb_subset_input_t *subset_input)
...
+
+/**
+ * hb_subset_input_drop_ot_layout:
+ * @subset_input: a subset_input.
+ *
+ * If enabled ot layout tables will be dropped as part of
+ * the subsetting operation. Currently this defaults to
+ * true.
+ *
+ * REPLACEME
```
Comment (ebraminio) [inline src/hb-subset-input.cc]: `Since: REPLACE` would be better for the next time :)

Comment (garretrieger) [reply src/hb-subset-input.cc]: Fixed.

File: src/hb-subset-input.cc [hunk 3]
```diff
@@ -45,6 +45,7 @@ hb_subset_input_create_or_fail (void)
 
   input->unicodes = hb_set_create ();
   input->glyphs = hb_set_create ();
+  input->drop_ot_layout = true;
```
[the remark this answers is not in the document]
Comment (ebraminio) [reply src/hb-subset-input.cc]: Or, the bot is drunk! (I don't see this reasonable)

[the remark this answers is not in the document]
Comment (garretrieger) [reply src/hb-subset-input.cc]: Agreed.

**Pull request merged**